crates/sui-indexer-alt-framework/src/lib.rs

```diff
@@ -284,10 +290,14 @@ impl<S: Store> Indexer<S> {
         }))
     }
 
-    /// Update the indexer's first checkpoint based on the watermark for the pipeline by adding for
-    /// handler `H` (as long as it's enabled). Returns `Ok(None)` if the pipeline is disabled,
-    /// `Ok(Some(None))` if the pipeline is enabled but its watermark is not found, and
-    /// `Ok(Some(Some(watermark)))` if the pipeline is enabled and the watermark is found.
+    /// Add a pipeline to the indexer and retrieve its watermark from the database to update the
```

Comment:
minor nit: I would not recommend documenting what a function's return value is for (only how it relates to its inputs etc) -- it kind of defeats the purpose of abstracting it, and it will quickly become untrue as the abstraction is used for other things.